syntax = "proto3";

package proto;

message Outfit {
  string name = 1;
  repeated uint32 armors = 2; // A list of formIDs that will be resolved as pointers to RE::TESObjectARMO
  bool is_favorite = 3;
}

message ActorOutfitAssignment {
  string current_outfit_name = 1;
  map<uint32, string> location_based_outfits = 2;
}

message OutfitSystem {
  bool enabled = 1;
  repeated Outfit outfits = 2;
  map<uint64, ActorOutfitAssignment> actor_outfit_assignments = 3;
}
